omod/src/main/java/org/openmrs/web/servlet/LoginServlet.java

```diff
@@ -178,12 +180,24 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 			catch (ContextAuthenticationException e) {
 				// set the error message for the user telling them
 				// to try again
-				httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "auth.password.invalid");
+				Integer maximumAlowedAttempts = 7;
+				if (loginAttemptsByUser <= maximumAlowedAttempts) {
+					Integer remainingAttempts = maximumAlowedAttempts - loginAttemptsByUser;
+					httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR,
+						"Invalid username/password. Please try again." + " " + remainingAttempts + " Attempts remaining");
+				}
+				
+				if (loginAttemptsByUser > maximumAlowedAttempts) {
+					
+					httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR,
+							"You have attempted to log in too many times and have been Locked out. Please try again later in 5 minutes");
```

Comment:
What happens for users who are using other non English locales?

Reply:
i actually really hadnt considered the issue of non english locale.